pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -273,14 +274,13 @@ static PyObject *get_sub_attr(PyObject *obj, char *attr, char *subAttr) {
 }
 
 static int is_simple_frame(PyObject *obj) {
-    PyObject *check = get_sub_attr(obj, "_mgr", "is_mixed_type");
-    int ret = (check == Py_False);
-
-    if (!check) {
+    PyObject *mgr = PyObject_GetAttrString(obj, "_mgr");
+    if (!mgr) {
         return 0;
     }
+    int ret = (get_attr_length(mgr, "blocks") <= 1);
```

Comment:
Hmm I'm not really sure about this change. The problem with how this code is currently set up is that when `is_simple_frame` is True then we will just serialize whatever `.values` is for that DataFrame. However, the rules for extension arrays are a bit different.

Do we have test coverage for DataFrames that only contain 1 block of an extension array? If not I'd be worried this could change things that we aren't testing currently

Reply:
My motivation here was to replace the `is_mixed_type` property the same way it was handled in #40525 (so theoretically this should revert behavior to pre #40525).  

Will look into your question and see if that case is covered

Reply:
Didn't see any coverage, have added some simple tests for the one `ExtensionBlock` case (which match 1.1.5 behavior). Let me know if this is what you had in mind!